frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx

```diff
@@ -718,6 +775,12 @@ describe("QueryBuilder", () => {
           expect(
             screen.queryByTestId("leave-confirmation"),
           ).not.toBeInTheDocument();
+
+          history.goBack();
+
+          expect(
+            screen.queryByTestId("leave-confirmation"),
+          ).not.toBeInTheDocument();
```

Comment:
This `history.goBack()` + assertion were added to each test that saves changes.
It follows the `does not show custom warning modal after editing model-based question via notebook editor and saving it` test case, which is also valid here (although it always worked).